src/Host/Client/Impl/Host/BrokerClient.cs

```diff
@@ -87,10 +88,18 @@ internal abstract class BrokerClient : IBrokerClient {
         protected abstract void OnCredentialsValidated(bool isValid);
 
         public async Task PingAsync() {
-            try {
-                (await HttpClient.PostAsync("/ping", new StringContent(""))).EnsureSuccessStatusCode();
-            } catch (HttpRequestException ex) {
-                throw new RHostDisconnectedException(Resources.Error_HostNotResponding.FormatInvariant(ex.Message), ex);
+            if (HttpClient != null) {
+                var isOnline = await CheckMachineOnlineAsync();
```

Comment:
PingAsync is a bit of a perf hit so I didn't want to put it in the high traffic area. But sure.

---
In reply to: [81587401](https://github.com/Microsoft/RTVS/pull/2416#discussion_r81587401) [](ancestors = 81587401,81587164)